File: python/transmit/resources/deliveries.py
from typing import Any, Dict, List


class Deliveries:
    """Deprecated — use ``Orders`` instead."""

    def __init__(self, client):
        self._client = client

    def create(self, data: Dict[str, Any]) -> Dict[str, Any]:
        return self._client.post("/api/v1/delivery-orders", json=data)

    def list(self, params: Dict[str, Any] = None) -> List[Dict[str, Any]]:
        return self._client.get("/api/v1/delivery-orders", params=params)

    def retrieve(self, delivery_id: str) -> Dict[str, Any]:
        return self._client.get(f"/api/v1/delivery-orders/{delivery_id}")

    def quote(self, data: Dict[str, Any]) -> Dict[str, Any]:
        return self._client.post("/api/v1/delivery-orders/calculate-fare", json=data)

File: go/transmit/deliveries.go
package transmit

import (
	"encoding/json"
	"fmt"
)

type DeliveriesService struct {
	client *Client
}

// Deprecated: use OrdersService
func (s *DeliveriesService) Create(req *CreateDeliveryOrderRequest) (map[string]interface{}, error) {
	resp, err := s.client.sendRequest("POST", "/api/v1/delivery-orders", req)
	if err != nil {
		return nil, err
	}
	var result map[string]interface{}
	err = json.Unmarshal(resp, &result)
	return result, err
}

func (s *DeliveriesService) List() ([]map[string]interface{}, error) {
	resp, err := s.client.sendRequest("GET", "/api/v1/delivery-orders", nil)
	if err != nil {
		return nil, err
	}
	var result []map[string]interface{}
	err = json.Unmarshal(resp, &result)
	return result, err
}

func (s *DeliveriesService) Retrieve(id string) (map[string]interface{}, error) {
	path := fmt.Sprintf("/api/v1/delivery-orders/%s", id)
	resp, err := s.client.sendRequest("GET", path, nil)
	if err != nil {
		return nil, err
	}
	var result map[string]interface{}
	err = json.Unmarshal(resp, &result)
	return result, err
}

func (s *DeliveriesService) Quote(req *CalculateOrderFareRequest) (map[string]interface{}, error) {
	resp, err := s.client.sendRequest("POST", "/api/v1/delivery-orders/calculate-fare", req)
	if err != nil {
		return nil, err
	}
	var result map[string]interface{}
	err = json.Unmarshal(resp, &result)
	return result, err
}
